Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +27,51 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+
+    return totalAmpHour_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+
+    if(current_<0){
+       return true;
+    }
     return false;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
-{
-    return QTime::currentTime();
+{    QTime charging;
```

Comment:
Put `QTime charging` on a new line separate from the brace

Reply:
Also, this isn't the most intuitive name - what if the battery isn't charging? Think about what this value is supposed to represent.